src/RouteCollector.php

```diff
@@ -146,36 +160,28 @@ public function getRoutes() : array
      *
      * @throws Exception\DuplicateRouteException on duplicate route detection.
      */
-    private function checkForDuplicateRoute(string $path, array $methods = null) : void
+    private function checkForDuplicateRoute(Route $route) : void
     {
-        if (null === $methods) {
-            $methods = Route::HTTP_METHOD_ANY;
+        if (array_key_exists($route->getName(), $this->routesNames)) {
```

Comment:
Why not `isset` here and below? It should be even faster ! :)

Reply:
fixed, but array_key_exists vs isset really is micro optimization

Reply:
If the value when set can never be a false-y value, `isset` is simpler to read, _and_ faster. It's the combination of benefits that makes it a recommended approach.

If there is a chance the value could be false-y, `array_key_exists()` is the appropriate solution, to prevent false negative lookups.